src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -305,9 +307,12 @@ private void WatchForFileSystemEventsThreadStart(object arg)
 
                 for (long i = 0; i < numEvents.ToInt32(); i++)
                 {
+                    Debug.Assert(eventPaths[i].Length > 0, "Null events are not supported");
+                    Debug.Assert(eventPaths[i][eventPaths[i].Length - 1] != '/', "Trailing slashes on events is not supported");
+
                     // Match Windows and don't notify us about changes to the Root folder
                     string path = eventPaths[i];
-                    if (path.Equals(_fullDirectory, StringComparison.OrdinalIgnoreCase))
+                    if (string.Compare(path, 0, _fullDirectory, 0, path.Length, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Is it possible the path could be a valid subset of full directory, e.g. if the event path is actually a parent of full directory rather than a sub directory?  This would compare successfully in that case.